src/opera.jl

```diff
@@ -1,72 +1,262 @@
+A dynamic structure that 
+ - contains a **directory of algebraic agents** (dictionary of `uuid => agent` pairs);
+ - keeps track of, and executes, **futures (delayed interactions)**;
+ - keeps track of, and executes, **system controls**;
+ - keeps track of, and executes, **instantious interactions**;
 
-# Algebraic Interactions
-It is possible to schedule additional interactions within the complex;
-such actions are instances of `AbstractOperaCall`,
-and they are modeled as tuples `(priority=0., call)`.
+# Futures
+You may schedule function calls, to be executed at predetermined points of time.
+The action is specified as a tuple `(id, call, time)`, where `id` is an optional textual identifier of the action, `call` is a (parameterless) anonymous function, which will be called at given `time`.
+Once the action is executed, the return value with corresponding action id and execution time is added to `futures_log` field of `Opera` instance.
 
-At the end of the topmost call to `step!`, the actions will be executed one-by-one in order of the respective priorities.
+See [`add_future!`](@ref) and [`@future`](@ref).
 
-In particular, you may schedule interactions of two kinds:
- 
- - `@schedule agent priority=0`, which will translate into a call `_interact!(agent)`,
- - `@schedule_call agent f(args...) priority=0` or `@schedule_call agent x->ex priority=0`,
-which will translate into a call `agent->f(agent, args...)` or `(x->ex)(agent)`, respectively.
+# Control Interactions
+You may schedule control function calls, to be executed at every step of the model.
+The action is specified as a tuple `(id, call)`, where `id` is an optional textual identifier of the action, and `call` is a (parameterless) anonymous function.
+Once the action is executed, the return value with corresponding action id and execution time is added to `controls_log` field of `Opera` instance.
+
+See [`add_control!`](@ref) and [`@control`](@ref).
 
-See [`@schedule`](@ref) and [`@schedule_call`](@ref).
+# Instantious Interactions
+You may schedule additional interactions which exist within a single step of the model;
+such actions are modeled as named tuples `(priority=0., call)`. Here, `call` is a (parameterless) anonymous function.
 
 They exist within a single step of the model and are executed after the calls
-to `_prestep!` and `_step!` finish. 
+to `_prestep!` and `_step!` finish.
+
+In particular, you may schedule interactions of two kinds:
+ 
+ - `poke(agent)`, which will translate into a call `_interact!(agent)`,
+ - `@call opera expresion priority=0`, which will translate into a call `() -> expression`.
 
-See [`opera_enqueue!`](@ref).
+See [`poke`](@ref) and [`@call`](@ref).
 """
 mutable struct Opera
-    calls::PriorityQueue{AbstractOperaCall, Float64}
+    # dictionary of `uuid => agent` pairs
     directory::Dict{UUID, AbstractAlgebraicAgent}
+    # intantious interactions
+    instantious_interactions::Vector{InstantiousInteraction}
```

Comment:
Yes, there was a priority queue before - but I thought that a simple vector would suffice here (sorted inserts cause a minimal overhead).

Reply:
Ok, makes sense. Maybe it would be nice to move the sorting code (the while loop) out into its own function since it is shared between `add_instantious_interaction!` and `add_future!` (I think), and that way it can be independently tested.